volatility3/framework/symbols/windows/extensions/mbr.py

```diff
@@ -0,0 +1,93 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import struct
+
+from volatility3.framework import objects
+
+class PARTITION_TABLE(objects.StructType):
+
+    def get_disk_signature(self) -> str:
+        """Get Disk Signature (GUID)."""
+        return "{0:02x}-{1:02x}-{2:02x}-{3:02x}".format(
+                self.DiskSignature[0], 
+                self.DiskSignature[1],
+                self.DiskSignature[2],
+                self.DiskSignature[3]
+        )
+
+class PARTITION_ENTRY(objects.StructType):
+
+    def set_index(self, index:int):
+        """Set Partition Entry Index."""
```

Comment:
This all looks really good, very glad of the docstrings and the type info!  5:D

Reply:
Thank you for your comment, some code in the file will be changed as the part processed by Enumeration is removed!